codecs/option.ts

```diff
@@ -1,36 +1,36 @@
 import { Codec, createCodec, metadata, ScaleDecodeError } from "../common/mod.ts"
 
-export function option<Some>($some: Codec<Some>): Codec<Some | undefined> {
+export function option<T>($some: Codec<T>): Codec<T | undefined>
+export function option<T, U>($some: Codec<T>, none: U): Codec<T | U>
+export function option<T, U>($some: Codec<T>, none?: U): Codec<T | U> {
   if ($some._metadata.some((x) => x.factory === option)) {
     throw new Error("Nested option codec will not roundtrip correctly")
   }
   return createCodec({
-    _metadata: metadata("$.option", option, $some),
+    _metadata: metadata("$.option", option<T, U>, $some, none!),
```

Comment:
```suggestion
    _metadata: metadata("$.option", option<T, U>, $some, ...none === undefined ? [] as never : [none!]),
```

Reply:
This unfortunately caused a type error. Do you foresee any issues with usage of `none!`?

Reply:
Yes it makes the printed metadata look terrible

Reply:
Any concerns about this solution?: just wrapped the entire element in parens and `never`ed it. Successfully suppresses the type error.

Reply:
`as never as [U]`